Web/Scotty/Internal/Types.hs

```diff
@@ -51,6 +52,16 @@ data Options = Options { verbose :: Int -- ^ 0 = silent, 1(def) = startup banner
 instance Default Options where
     def = Options 1 defaultSettings
 
+newtype RouteOptions = RouteOptions { maxRequestSize :: Maybe Kilobytes -- max allowed request size in KB
+                                    }
+
+instance Default RouteOptions where
+    def = RouteOptions Nothing
+
+mkRouteOptions :: Kilobytes -> RouteOptions
+mkRouteOptions = RouteOptions . Just
```

Comment:
Can  we do the simple thing and export the `RouteOptions` constructor instead of this indirection? It's in `Internal`, so I think people understand it's subject to change. This one, itself is subject to change, since we need to add parameters if the number of fields grow, so this indirection doesn't really buy us any flexibility. In fact, we'd get _more_ flexibility using the built-in record syntax.

Reply:
If you're dead-set against exporting the constructor (I don't get why, but whatever), can you please replace this with a `setMexRequestSize :: Kilobytes -> RouteOptions -> RouteOptions`?

Reply:
haha I am not dead-set against anything, I think there are always multiple (competing) viable options in the design space :) 

I removed "mkRouteOptions" as it did not buy flexibility in case of code changes as you point out. I didn't want to introduce RouteOptions to the outside (we can discuss! :) ) so I added a [function](https://github.com/hotlib/scotty/blob/4d89f8f587050503ced7ce20382e5de457cd5525/Web/Scotty/Internal/Types.hs#L89) that updates only a specific property (max. request-body size in this case). This way we can for now perceive RouteOptions as just a internal thing without the need to expose it.

Reply:
But I am totally open to any suggestions!